### PR TITLE
Running tests on localstack

### PR DESCRIPTION
**Pull request by fabricio-aguiar**
https://pulp.plan.io/issues/6155
ref #6155

### CODE REVIEW

**Comment by fabricio-aguiar** [comment]
@bmbouter @daviddavis @dkliban could you please give me some feedback? Once I get an approval, I'll move these changes to plugin_template

**Comment by dkliban** [comment]
The tests should run on PRs and Cron jobs. We should run jut the pulp_file tests.

File: pulp_file/tests/functional/utils/__init__.py [hunk 1]
```diff
@@ -173,8 +177,10 @@ def monitor_task(task_href):
     completed = ["completed", "failed", "canceled"]
     task = tasks.read(task_href)
     while task.state not in completed:
-        sleep(2)
+        sleep(1)
         task = tasks.read(task_href)
+        if "S3Boto3Storage" in settings.DEFAULT_FILE_STORAGE:
+            create_bucket()
```
Comment (daviddavis) [inline pulp_file/tests/functional/utils/__init__.py]: Why are we creating buckets here?

Comment (fabricio-aguiar) [reply pulp_file/tests/functional/utils/__init__.py]: For minio, I have some problems with the connections pool:
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
travis: https://travis-ci.org/pulp/pulp_file/jobs/654626953?utm_medium=notification&utm_source=github_status
code: https://github.com/pulp/pulp_file/pull/358/files

For localstack, I can see the bucket has some lifetime, and it dies before the tests finish, so I added a conftest, and this method here, for keeping the bucket alive.

Comment (fabricio-aguiar) [reply pulp_file/tests/functional/utils/__init__.py]: In theory:
- localstack bucket should not die, as I set `DATA_DIR`
- minio problem is fixed: https://github.com/minio/minio/pull/6415
But for both, I could not make it work locally and here on travis, the only way I could make localstack work was the trick of keeping creating bucket

Comment (daviddavis) [reply pulp_file/tests/functional/utils/__init__.py]: There's no way to configure the lifetime of buckets in localstack? That also seems suboptimal for development as well if I have to create a bucket every so often.

**Comment by fabricio-aguiar** [comment]
closing in favor of minio: https://github.com/pulp/pulp_file/pull/358

**Pull request closed without merging**